Moller-Plesset/UMP2_Spin_Adapted.py

```diff
@@ -0,0 +1,161 @@
+"""
+A reference implementation of MP2 from a UHF reference in the spatial orbital basis.
+Intended to show basic example of spin adapting a spin orbital code.
+"""
+
+__authors__ = "Matthew McAllister Davis"
+__credits__ = ["Matthew McAllister Davis", "Justin M. Turney"]
+__copyright__ = "(c) 2014-2020, The Psi4NumPy Developers"
+__license__ = "BSD-3-Clause"
+__date__ = "2020-01-08"
+
+import psi4
+import numpy as np
+from time import time
+np.set_printoptions(precision=10, linewidth=200, suppress=True)
+psi4.core.be_quiet()
+psi4.core.set_output_file("output.dat")
+psi4.set_memory('2 GB')
+
+# Computing on ethane cation!
+mol = psi4.geometry ( """
+        1 2
+        C            0.080813391216    -0.780500814497     0.000000070444
+        C           -0.080813391216     0.780500814497     0.000000070444
+        H            0.459851209402    -1.219923677395    -0.928858893056
+        H           -1.070346469382    -0.754016692936    -0.000002364939
+        H            0.459848034953    -1.219923498861     0.928860419230
+        H           -0.459851209402     1.219923677395    -0.928858893056
+        H            1.070346469382     0.754016692936    -0.000002364939
+        H           -0.459848034953     1.219923498861     0.928860419230
+        symmetry c1
+        """)
+
+compare = True
+psi4.set_options({'basis':'cc-pvdz',
+                  'reference':'uhf',
+                  'mp2_type':'conv',
+                  'e_convergence':1e-12,
+                  'd_convergence':1e-10,
+                  'scf_type':'pk'})
+
+# Get SCF wavefunction and basic info
+print('Starting RHF.')
+t = time()
+e,wfn = psi4.energy('scf',return_wfn=True,mol=mol)
+t = time() - t
+print(f'Computed @RHF {e:16.10f} Eh in {t:4.6f} s.\n')
+nmo = wfn.nmo()
+nalpha = wfn.nalpha()
+nbeta = wfn.nbeta()
+
+# Core hamiltonian in AO basis
+h = wfn.H().np
+
+# Orbital eigenvalues for MP2 denominators
+epsa = wfn.epsilon_a().np
+epsb = wfn.epsilon_b().np
+
+# Make a MintsHelper for integral transformation
+mints = psi4.core.MintsHelper(wfn.basisset())
+
+# size of occ and vir spaces for alpha and beta spin
+occa = nalpha
+vira = nmo - nalpha
+occb = nbeta
+virb = nmo - nbeta
+
+# occ and vir slices for alpha and beta
+oa = slice(0,nalpha)
+va = slice(nalpha,nmo)
+ob = slice(0,nbeta)
+vb = slice(nbeta,nmo)
+
+# AO->MO coefficients, all
+Ca = wfn.Ca().np
+Cb = wfn.Cb().np
```

Comment:
Unless I'm mistaken `Ca` and `Cb` are not used anywhere; likewise, for `ha` and `hb` below.  If that's the case, let's remove them to avoid any confusion.

Reply:
:+1: done